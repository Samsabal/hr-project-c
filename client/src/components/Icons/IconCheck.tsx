import React from 'react'
import { IconProps } from '../../types/IconProps'

export const IconCheck = ({ size, fill, color, strokeWidth = 2 }: IconProps) => {
    return (
        <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
            <path d="M20 6L9 17L4 12" strokeWidth={strokeWidth} strokeLinecap="round" strokeLinejoin="round" />
        </svg>
    )
}
