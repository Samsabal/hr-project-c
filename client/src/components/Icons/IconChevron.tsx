import React from 'react'

import { IconProps } from '../../types/IconProps'

export const IconChevron = ({ size, fill, color, strokeWidth = 2, direction }: IconProps) => {
    switch (direction) {
        case 'down':
            return (
                <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
                    <path d="M6 9L12 15L18 9" strokeWidth={strokeWidth} strokeLinecap="round" strokeLinejoin="round" />
                </svg>
            );
        case 'up':
            return (
                <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
                    <path d="M18 15L12 9L6 15" strokeWidth={strokeWidth} strokeLinecap="round" strokeLinejoin="round" />
                </svg>
            );
    }
    // Return a chevron pointing right by default
    return (
        <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
            <path d="M9 18L15 12L9 6" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" />
        </svg>
    );
}