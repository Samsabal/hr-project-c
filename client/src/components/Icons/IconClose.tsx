import React from 'react'
import { IconProps } from '../../types/IconProps'

export const IconClose = ({ size, fill, color, strokeWidth = 2 }: IconProps) => {
    return (
        <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
            <path d="M18 6L6 18M6 6L18 18" strokeWidth={strokeWidth} strokeLinecap="round" strokeLinejoin="round" />
        </svg>
    )
}
