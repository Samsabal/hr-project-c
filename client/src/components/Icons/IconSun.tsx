import React from 'react'
import { IconProps } from '../../types/IconProps'

export const IconSun = ({ size, fill = 'none', color, strokeWidth = 2 }: IconProps) => {
    return (
        <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
            <path opacity="0.12" d="M12 17C14.7614 17 17 14.7614 17 12C17 9.23858 14.7614 7 12 7C9.23858 7 7 9.23858 7 12C7 14.7614 9.23858 17 12 17Z" className={fill} />
            <path d="M12 2V4M12 20V22M4 12H2M6.31412 6.31412L4.8999 4.8999M17.6859 6.31412L19.1001 4.8999M6.31412 17.69L4.8999 19.1042M17.6859 17.69L19.1001 19.1042M22 12H20M17 12C17 14.7614 14.7614 17 12 17C9.23858 17 7 14.7614 7 12C7 9.23858 9.23858 7 12 7C14.7614 7 17 9.23858 17 12Z" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" />
        </svg>
    )
}
