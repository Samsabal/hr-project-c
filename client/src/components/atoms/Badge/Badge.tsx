import React from "react";

type BadgeProps = {
    size: "sm" | "md" | "lg";
    color: "gray" | "primary" | "error" | "success";
    icon?: JSX.Element;
    text: string;
};

export function Badge({ size, color, icon, text }: BadgeProps) {
    let badgeSizeStyle = "";
    let badgeColorStyle = "";

    switch (size) {
        case "sm":
            badgeSizeStyle = "py-0.5 pl-1.5 pr-2 text-xs";
            break;
        case "md":
            badgeSizeStyle = "py-0.5 pl-2 pr-2.5 text-sm";
            break;
        case "lg":
            badgeSizeStyle = "py-1 pl-2.5 pr-3 text-sm";
            break;
    }

    switch (color) {
        case "gray":
            badgeColorStyle = "text-gray-700 bg-gray-100 dark:bg-gray-400 dark:bg-opacity-10 dark:text-gray-300";
            break;
        case "primary":
            badgeColorStyle = "text-primary-500 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10";
            break;
        case "error":
            badgeColorStyle = "text-error-500 bg-error-50 dark:bg-error-400 dark:text-error-400 dark:bg-opacity-10";
            break;
        case "success":
            badgeColorStyle = "text-success-500 bg-success-50 dark:bg-success-400 dark:bg-opacity-10";
            break;
    }

    return (
        <div className={`${badgeSizeStyle} ${badgeColorStyle} flex rounded-full font-medium items-center gap-1`}>
            {icon}
            <span>{text}</span>
        </div>
    );
}
