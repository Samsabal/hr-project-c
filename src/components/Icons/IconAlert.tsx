import React from 'react'
import { IconProps } from '../../types/IconProps'

export const IconAlert = ({ size, fill, color, strokeWidth = 2 }: IconProps) => {
    return (
        <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill={"none"}>
            <path className={fill} opacity="0.12" d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22Z" />
            <path d="M12 8V12M12 16H12.01M22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12Z" strokeWidth={strokeWidth} strokeLinecap="round" strokeLinejoin="round" />
        </svg>

    )
}
