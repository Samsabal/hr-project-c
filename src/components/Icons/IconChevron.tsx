import React from 'react'

import { IconProps } from '../../types/IconProps'

export const IconChevron = ({ size, fill, color, strokeWidth = 2, direction }: IconProps) => {
    switch (direction) {
        case 'right':
            return (
                <svg className={color} width={size}height={size} viewBox="0 0 24 24" fill="none">
                    <path d="M9 18L15 12L9 6" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" />
                </svg>
            );
    }
    // Return a chevron pointing right by default
    return (
        <svg className={color} width={size}height={size} viewBox="0 0 24 24" fill="none">
            <path d="M9 18L15 12L9 6" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" />
        </svg>
    );
}