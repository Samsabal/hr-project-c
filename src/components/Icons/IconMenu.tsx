import React from 'react'
import { IconProps } from '../../types/IconProps'

export const IconMenu = ({ size, fill, color, strokeWidth = 2 }: IconProps) => {
    return (
        <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
            <path d="M3 12H21M3 6H21M9 18H21" strokeWidth={strokeWidth} strokeLinecap="round" strokeLinejoin="round" />
        </svg>
    )
}