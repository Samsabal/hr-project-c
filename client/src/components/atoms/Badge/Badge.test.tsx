import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import '@testing-library/jest-dom'
import { Badge } from "./Badge";
import { IconPlus } from "../Icons/Icons";

describe ("Badge", () => {
    test('renders badge component correctly and has only required props', () => {
        const container1 = render(
            <Badge
            size="sm"
            color="primary"
            text="Badge"
            />
        );
        expect(container1).toMatchSnapshot();
        
        const container2 = render(
            <Badge
            size="md"
            color="success"
            text="Badge"
            />
        );
        expect(container2).toMatchSnapshot();
        
        const container3 = render(
            <Badge
            size="lg"
            color="error"
            text="Badge"
            />
        );
        expect(container3).toMatchSnapshot();
    });

    test('renders badge component correctly, has required props and tests optional props', () => {
        const { container } = render(
            <Badge
            size="sm"
            color="gray"
            text="Badge"
            icon={<IconPlus size="10" color="stroke-gray-700 dark:stroke-white" fill="" />}
            />
        );
        expect(container).toMatchSnapshot();
    });
});