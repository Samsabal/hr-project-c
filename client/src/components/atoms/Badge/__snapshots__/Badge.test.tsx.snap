// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders badge component correctly and has only required props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="py-0.5 pl-1.5 pr-2 text-xs text-primary-600 dark:text-primary-500 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 flex w-max rounded-full font-medium items-center gap-1"
      >
        <span>
          Badge
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="py-0.5 pl-1.5 pr-2 text-xs text-primary-600 dark:text-primary-500 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 flex w-max rounded-full font-medium items-center gap-1"
    >
      <span>
        Badge
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Badge renders badge component correctly and has only required props 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="py-0.5 pl-1.5 pr-2 text-xs text-primary-600 dark:text-primary-500 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 flex w-max rounded-full font-medium items-center gap-1"
      >
        <span>
          Badge
        </span>
      </div>
    </div>
    <div>
      <div
        class="py-0.5 pl-2 pr-2.5 text-sm text-success-500 bg-success-50 dark:bg-success-400 dark:bg-opacity-10 flex w-max rounded-full font-medium items-center gap-1"
      >
        <span>
          Badge
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="py-0.5 pl-2 pr-2.5 text-sm text-success-500 bg-success-50 dark:bg-success-400 dark:bg-opacity-10 flex w-max rounded-full font-medium items-center gap-1"
    >
      <span>
        Badge
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Badge renders badge component correctly and has only required props 3`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="py-0.5 pl-1.5 pr-2 text-xs text-primary-600 dark:text-primary-500 bg-primary-50 dark:bg-primary-400 dark:bg-opacity-10 flex w-max rounded-full font-medium items-center gap-1"
      >
        <span>
          Badge
        </span>
      </div>
    </div>
    <div>
      <div
        class="py-0.5 pl-2 pr-2.5 text-sm text-success-500 bg-success-50 dark:bg-success-400 dark:bg-opacity-10 flex w-max rounded-full font-medium items-center gap-1"
      >
        <span>
          Badge
        </span>
      </div>
    </div>
    <div>
      <div
        class="py-1 pl-2.5 pr-3 text-sm text-error-500 bg-error-50 dark:bg-error-400 dark:text-error-400 dark:bg-opacity-10 flex w-max rounded-full font-medium items-center gap-1"
      >
        <span>
          Badge
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="py-1 pl-2.5 pr-3 text-sm text-error-500 bg-error-50 dark:bg-error-400 dark:text-error-400 dark:bg-opacity-10 flex w-max rounded-full font-medium items-center gap-1"
    >
      <span>
        Badge
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Badge renders badge component correctly, has required props and tests optional props 1`] = `
<div>
  <div
    class="py-0.5 pl-1.5 pr-2 text-xs text-gray-700 bg-gray-100 dark:bg-gray-400 dark:bg-opacity-10 dark:text-gray-300 flex w-max rounded-full font-medium items-center gap-1"
  >
    <svg
      class="stroke-gray-700 dark:stroke-white"
      fill="none"
      height="10"
      viewBox="0 0 24 24"
      width="10"
    >
      <path
        d="M12 5V19M5 12H19"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
    <span>
      Badge
    </span>
  </div>
</div>
`;
